src/Pages/FieldInfoPage.js

```diff
@@ -0,0 +1,244 @@
+import {
+  Button,
+  FormControl,
+  FormGroup,
+  InputLabel,
+  MenuItem,
+  Select,
+  Snackbar,
+  Typography,
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import { Alert } from '@material-ui/lab';
+import React, { useState, useEffect } from 'react';
+import { getAllFields } from 'dataclients/FieldsClient';
+import { getLocations } from 'dataclients/LocationsClient';
+import TableComponent from '../components/common/TableComponent';
+
+/**
+ * css styles for Field Info Page
+ */
+const useStyles = makeStyles((theme) => ({
+  formControl: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+    marginLeft: 'auto',
+    marginRight: 'auto',
+    minWidth: 250,
+    maxWidth: 300,
+  }
+}));
+
+
+/**
+ * Define columns for Fields Info Table
+ * * Need to add last two entries for edit and delete icons to be visible
+ */
+const columnData = [
+  { id: 'id', label: 'Id', minWidth: 5 },
+  { id: 'name', label: 'Field Name', minWidth: 30 },
+  {
+    id: 'location',
+    label: 'Locality',
+    minWidth: 30
+  },
+  {
+    id: 'area',
+    label: 'Area',
+    minWidth: 10
+  },
+  {
+    id: 'edit',
+    type: 'icon',
+    minWidth: 5,
+    align: 'left',
+    label: ''
+  },
+  {
+    id: 'delete',
+    type: 'icon',
+    minWidth: 5,
+    align: 'left',
+    label: ''
+  }
+];
+
+/**
+ * Component to show Field Information
+ * @returns form allowing location selection
+ * @returns Table to show field info
+ */
+function FieldInfoPage() {
+  const classes = useStyles();
+  const [locations, setLocations] = useState([]);
+  const [locality, setLocality] = useState('');
+  const [alertMessage, setAlertMessage] = useState('');
+  const [alertStatus, setAlertStatus] = useState(false);
+  const [alertSeverity, setAlertSeverity] = useState('');
+  const [fieldsData, setFields] = useState([]);
+
+  useEffect(() => {
+    getLocations()
+      .then(setLocations)
+      .catch((e) => {
+        console.log('Fetching all locations failed', e);
```

Comment:
Please add an alert here as well for failure.

Reply:
done